include/openthread/platform/border_routing.h

```diff
@@ -65,6 +65,23 @@ extern "C" {
  */
 extern void otPlatBorderRoutingProcessIcmp6Ra(otInstance *aInstance, const uint8_t *aMessage, uint16_t aLength);
 
+/**
+ * Process a prefix received from the DHCPv6 PD Server. The prefix is received on
+ * the DHCPv6 PD client callback and provided to the Routing Manager via this
+ * API.
+ *
+ * The lifetime of the prefix can be updated by calling the function again with updated
+ * time values.
```

Comment:
Some questions about the API document

- How to deprecate a prefix?

For PD, to deprecate a prefix is to pass 0 as its lifetime, wouls suggest pointing this out in the API document.

- What if this function is called twice with different prefix?

For PD, we will more prefer GUA and a prefix with longer lifetime. Also suggest adding this to the document.

Reply:
I Updated the API description. 

One comment I would make is that the original implementation, that this PR updates, doesn't seem to prefer a prefix with a longer lifetime. I only see a check on the prefix length to choose a GUA instead of a ULA.